W5500.device.nut

```diff
@@ -0,0 +1,2572 @@
+    // **************************************************************************
+    function readReg(addr, bsb) {
+        local b = blob();
+        local cp = bsb | W5500_READ_COMMAND | W5500_VARIABLE_DATA_LENGTH;
+        local res = blob();
+
+        (_cs) ? _cs.write(0): _spi.chipselect(1);
+
+        b.writen(addr >> 8, 'b');
+        b.writen(addr & 0xFF, 'b');
+        b.writen(cp, 'b');
+        b.writen(0x00, 'b');
+
+        res = _spi.writeread(b);
+
+        (_cs) ? _cs.write(1): _spi.chipselect(0);
+
+        return res[3];
+    }
+
+    // ***************************************************************************
+    // writeReg
+    // Returns: null
+    // Parameters:
+    //      reg - register to write to
+    //      value - data to write to register
+    // **************************************************************************
+    function writeReg(addr, bsb, value) {
+        local b = blob();
+        local cp = bsb | W5500_WRITE_COMMAND | W5500_VARIABLE_DATA_LENGTH;
+
+        (_cs) ? _cs.write(0): _spi.chipselect(1);
+
+        b.writen(addr >> 8, 'b');
+        b.writen(addr & 0xFF, 'b');
+        b.writen(cp, 'b');
+        b.writen(value, 'b');
+
+        _spi.write(b);
+
+        (_cs) ? _cs.write(1): _spi.chipselect(0);
+    }
+
+
+
+    // PRIVATE FUNCTIONS
+    // ---------------------------------------------
+
+    // ***************************************************************************
+    // _hexToInt
+    // Returns: an integer representation of the string passed in
+    // Parameters:
+    //      hexString - A string representing a hexadecimal number
+    // **************************************************************************
+    function _hexToInt(hexString) {
+        // Does the string start with '0x'? If so, remove it
+        if (hexString.slice(0, 2) == "0x") hexString = hexString.slice(2);
+
+        // Get the integer value of the remaining string
+        local intValue = 0;
+
+        foreach (character in hexString) {
+            local nibble = character - '0';
+            if (nibble > 9) nibble = ((nibble & 0x1F) - 7);
+            intValue = (intValue << 4) + nibble;
+        }
+
+        return intValue;
+    }
+
+    // ***************************************************************************
+    // _addrToIP
+    // Returns: an array of four numbers representing an ip address or false if it is a string for the DNS
+    // Parameters:
+    //      addr - a string or array representing an IP address
+    // **************************************************************************
+    function _addrToIP(addr) {
+
+        if (typeof addr == "string" && addr.len() > 0) {
+
+            try {
+                local parts = split(addr, ".");
+                if (parts.len() == 4) {
+                    foreach (part in parts) {
+                        local ipart = part.tointeger();
+                        if (ipart.tostring() != part || ipart < 0 || ipart >= 2
```

Comment:
It was an attempt to reduce the polling to when it was "required" but it failed. I have removed it.
Hopefully when we come back and review the requirement for polling it will go away altogether.